x-pack/plugins/security_solution/server/lib/detection_engine/rules/bulk_actions/rule_params_modifier.ts

```diff
@@ -59,6 +71,10 @@ const applyBulkActionEditToRuleParams = (
         "Index patterns can't be overwritten. Machine learning rule doesn't have index patterns property"
       );
 
+      if (ruleParams.dataViewId != null && !action.overwrite_data_views) {
```

Comment:
Nit: maybe encapsulate this in a declarative helper? e.g. 


```suggestion
     if (actionNotValid(action, ruleParams)) {
```

Reply:
👍🏽 I think there's a different helper method @banderror had suggested we could use. Can definitely cleanup/revisit post 8.4.